pkg/expr/nodes.go

```diff
@@ -401,7 +409,44 @@ func (dn *DSNode) Execute(ctx context.Context, now time.Time, _ mathexp.Vars, s
 	}
 
 	var result mathexp.Results
-	responseType, result, err = s.converter.Convert(ctx, dn.datasource.Type, dataFrames, s.allowLongFrames)
+	// TODO: Don't run Datasource Node Results through converter if input is SQL expression
+	// Also for now, must be one frame and not of the fields may have labels
+	if dn.isSQLInput {
+		var convertForSQL bool
+		if len(dataFrames) > 1 {
+			convertForSQL = true
+		}
+
+		for _, frame := range dataFrames {
+			if convertForSQL {
+				break
+			}
+			for _, field := range frame.Fields {
```

Comment:
For me there a bit of mental gymnastics here.
We're already checked if there's more than one frame, and now we're checking if there's any field-labels, and we're doing it on the first (and only) frame. So could we make this easier to understand? Personally I find this _slightly_ nicer:

```
		// check the first frame for labels, if it exists
		if !convertForSQL && len(dataFrames) > 0 {
			for _, field := range dataFrames[0].Fields {
				if len(field.Labels) > 0 {
					convertForSQL = true
					break
				}
			}
		}
```

Reply:
Oh wait not you are correct, because of the lines above that i missed.